src/Symfony/Component/Workflow/Workflow.php

```diff
@@ -339,4 +358,59 @@ private function announce($subject, Transition $initialTransition, Marking $mark
             $this->dispatcher->dispatch(sprintf('workflow.%s.announce.%s', $this->name, $transition->getName()), $event);
         }
     }
+
+    /**
+     * @param string $transitionName
+     *
+     * @return Transition[]
+     */
+    private function getTransitionsByName(string $transitionName)
+    {
+        $transitions = array_filter(
+            $this->definition->getTransitions(),
+            function (Transition $transition) use ($transitionName) {
+                return $transition->getName() === $transitionName;
+            }
+        );
+
+        return $transitions;
+    }
+
+    /**
+     * Returns all enabled transitions or a transition blocker list of one of them.
+     *
+     * @param object $subject        A subject
+     * @param string $transitionName
+     *
+     * @return Transition[]|TransitionBlockerList All enabled transitions or a blocker list
+     *                                            if no enabled transitions can be found
+     */
+    private function getEnabledTransitionsByNameOrTransitionBlockerList($subject, string $transitionName)
```

Comment:
Very long name, I agree. This method achieves the same as:
```php
$transitions = $this->getEnabledTransitions();

if (!$transitions) {
  $transitionBlockerList = $this->whyCannot();
}
```
but the benefit is, that this method doesn't run the guards twice in case there are no enabled transitions.

This method wasn't needed until I found out, that unit tests, that test duplicated transition names cases, fail. I.e. this method is an example of the maintenance burden I mentioned in the PR.

Reply:
I really don't like this method either. Could you get ride of it ?

Reply:
I would like to get rid of it, but I can't because the only other solution is to have the guards be run twice in the pessimistic case scenario (i.e. when there are no enabled transitions). And I'd rather have an ugly but a very dedicated private method than runtime performance degradation.

I guess this is one of those cases, when one looks at other dev's code and thinks "I would've done it better", but when they try to rewrite it to make it better, they realize, there's no better solution.

Reply:
I would throw an exception that carries the list in case the transition is not enabled. this would be better instead of having multple return types

Reply:
I agree, But anyway, I'm currently rewrite the Workflow class ATM.
It has become too complex to understand. The very first version was very simple and now it too complex too me.
I will open a new PR today.